Twig/SecurityExtension.php

```diff
@@ -3,27 +3,33 @@
 namespace JMS\SecurityExtraBundle\Twig;
 
 use JMS\SecurityExtraBundle\Security\Authorization\Expression\Expression;
+use JMS\SecurityExtraBundle\Exception\InvalidArgumentException;
+use Symfony\Component\Security\Core\Authorization\AuthorizationChecker;
 use Symfony\Component\Security\Core\SecurityContextInterface;
 
 class SecurityExtension extends \Twig_Extension
 {
-    private $context;
+    private $authorizationChecker;
 
-    public function __construct(SecurityContextInterface $context)
+    public function __construct($authorizationChecker)
     {
-        $this->context = $context;
+        if (!$authorizationChecker instanceof SecurityContextInterface && !$authorizationChecker instanceof AuthorizationChecker) {
```

Comment:
@WouterJ was there a reason to not use the interface as suggested by @Ener-Getick and @stof? It seems like the better choice here.